### PR TITLE
[fix](catalog)paimon support more data type

### PR DESCRIPTION
**Pull request by zddr**
## Proposed changes

Issue Number: close #xxx

Support for more data types and predicate pushdown of more types

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: fe/be-java-extensions/paimon-scanner/src/main/java/org/apache/doris/paimon/PaimonColumnValue.java [hunk 1]
```diff
@@ -109,14 +109,13 @@ public byte[] getStringAsBytes() {
...
-        return Instant.ofEpochMilli(record.getTimestamp(idx, 3)
-                .getMillisecond()).atZone(ZoneOffset.ofHours(8)).toLocalDate();
+        return LocalDate.ofEpochDay(record.getLong(idx));
     }
 
     @Override
     public LocalDateTime getDateTime() {
         return Instant.ofEpochMilli(record.getTimestamp(idx, 3)
-            .getMillisecond()).atZone(ZoneOffset.ofHours(8)).toLocalDateTime();
+            .getMillisecond()).atZone(ZoneOffset.ofHours(0)).toLocalDateTime();
```
Comment (morningman) [inline fe/be-java-extensions/paimon-scanner/src/main/java/org/apache/doris/paimon/PaimonColumnValue.java:118]: Why changing to zero?

File: fe/fe-core/src/main/java/org/apache/doris/planner/external/paimon/PaimonValueConverter.java [hunk 2]
```diff
@@ -110,11 +121,15 @@ public Long visit(BigIntType bigIntType) {
...
+    // when a = 9.1,paimon can get data,doris can not get data
+    // when a > 9.1,paimon can not get data,doris can get data
+    // paimon is no problem,but we consistent with Doris internal table
+    // Therefore, comment out this code
     public Float visit(FloatType floatType) {
-        if (expr instanceof FloatLiteral) {
-            FloatLiteral floatLiteral = (FloatLiteral) expr;
-            return (float) floatLiteral.getValue();
-        }
+        //        if (expr instanceof FloatLiteral) {
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/planner/external/paimon/PaimonValueConverter.java:129]: remove unused code

File: fe/fe-core/src/main/java/org/apache/doris/planner/external/paimon/PaimonValueConverter.java [hunk 3]
```diff
@@ -127,39 +142,25 @@ public Double visit(DoubleType doubleType) {
     }
 
     public Object visit(DateType dateType) {
+        if (expr instanceof DateLiteral) {
+            DateLiteral dateLiteral = (DateLiteral) expr;
+            dateLiteral.getLongValue();
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/planner/external/paimon/PaimonValueConverter.java:147]: usused code?

**Comment by zddr** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 47.23 seconds
 stream load tsv:          508 seconds loaded 74807831229 Bytes, about 140 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          65 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          30 seconds loaded 861443392 Bytes, about 27 MB/s
 insert into select:          29.3 seconds inserted 10000000 Rows, about 341K ops/s
 storage size: 17162126358 Bytes

**Comment by zddr** [comment]
run buildall

**Review by morningman — APPROVED** [review]
LGTM

**Review by zy-kkk — APPROVED** [review]

**Pull request merged**